libs/Microsoft.MixedReality.WebRTC.Native/src/sdp_utils.h

```diff
@@ -55,4 +55,9 @@ std::string EncodeIceServers(const std::string& url,
                              const std::string& username,
                              const std::string& password);
 
+// Copied from src/pc/jsepsessiondescription.cc so that we don't rely on a binary
+// representation of absl::optional as per the Abseil Compatibility Guidelines:
+// https://abseil.io/about/compatibility
+std::optional<webrtc::SdpType> SdpTypeFromString(const std::string& type_str);
```

Comment:
We should revert to C++14 instead, which solves the issue with `std::optional`. That's what I did on the [experimental/undock](https://github.com/microsoft/MixedReality-WebRTC/tree/experimental/undock) branch, and it works fine, though this requires several changes to downgrade the Native projects to remove C++17 features. Not sure what's the best way to integrate those changes though.

Reply:
Your undock build instructions could integrate nicely into the /tools/build folder as another platform/target leveraging the new libwebrtc scripts.

Reply:
Is it worthwhile to add an 'invalid sdptype' so we can avoid the optional altogether?

Reply:
No, this needs to return a `webrtc::SdpType` to pass the value to `webrtc::CreateSessionDescription()` and that's a Google type we cannot change.

Also:
- on `master` we compile everything with C++17 so there is no discrepancy and `absl::optional` is an alias for `std::optional`
- on `experimental/undock` we reverted to C++14 for the interop DLL to be compatible with the `libwebrtc` code, and we use `absl::optional` only

...so we can then directly use the implementation in `jsepsessiondescription.cc` instead of making a copy of it. But I think @eanders-MS is possibly compiling some fork of `master` in C++14 (`libwebrtc`'s default) hence hitting this issue. This is a temporary problem.